addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/internal/KM200ErrorService.java

```diff
@@ -0,0 +1,146 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.openhab.binding.km200.internal;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+/**
+ * The KM200ErrorService representing a error service with its all capabilities
+ *
+ * @author Markus Eckhardt - Initial contribution
+ *
+ */
+public class KM200ErrorService {
+
+    private static final Logger logger = LoggerFactory.getLogger(KM200ErrorService.class);
+
+    private Integer activeError = 1;
+
+    /* List for all errors */
+    private final ArrayList<HashMap<String, String>> errorMap;
+
+    KM200ErrorService() {
+        errorMap = new ArrayList<HashMap<String, String>>();
+    }
+
+    /**
+     * This function removes all errors from the list
+     *
+     */
+    void removeAllErrors() {
+        synchronized (errorMap) {
+            if (errorMap != null) {
+                errorMap.clear();
+            }
+        }
+    }
+
+    /**
+     * This function updates the errors
+     *
+     */
+    void updateErrors(JsonObject nodeRoot) {
+        synchronized (errorMap) {
+            /* Update the list of errors */
+            try {
+                removeAllErrors();
+                JsonArray sPoints = nodeRoot.get("values").getAsJsonArray();
+                for (int i = 0; i < sPoints.size(); i++) {
+                    JsonObject subJSON = sPoints.get(i).getAsJsonObject();
+                    HashMap<String, String> valMap = new HashMap<String, String>();
+                    Set<Map.Entry<String, JsonElement>> oMap = subJSON.entrySet();
+                    oMap.forEach(item -> {
+                        logger.debug("Set: {} val: {}", item.getKey(), item.getValue().getAsString());
+                        valMap.put(item.getKey(), item.getValue().getAsString());
+                    });
+                    errorMap.add(valMap);
+                }
+            } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception

Reply:
Removed this try/catch here.